server/schedule/filter/filters.go

```diff
@@ -227,67 +227,80 @@ type StoreStateFilter struct {
 	MoveRegion bool
 	// Set true if allows temporary states.
 	AllowTemporaryStates bool
+	// Typ is used to distinguish the reason of store state filter
+	Typ string
 }
 
 // Scope returns the scheduler or the checker which the filter acts on.
-func (f StoreStateFilter) Scope() string {
+func (f *StoreStateFilter) Scope() string {
 	return f.ActionScope
 }
 
 // Type returns the type of the Filter.
-func (f StoreStateFilter) Type() string {
-	return "store-state-filter"
+func (f *StoreStateFilter) Type() string {
+	return f.Typ
 }
 
 // conditionFunc defines condition to determine a store should be selected.
 // It should consider if the filter allows temporary states.
 type conditionFunc func(*config.PersistOptions, *core.StoreInfo) bool
 
-func (f StoreStateFilter) isTombstone(opt *config.PersistOptions, store *core.StoreInfo) bool {
+func (f *StoreStateFilter) isTombstone(opt *config.PersistOptions, store *core.StoreInfo) bool {
```

Comment:
you can change it to `isTombstone(opt *config.PersistOptions, store *core.StoreInfo) (reason string, ok bool)`

Reply:
The reason I use a pointer receiver here is to keep consistent with other filters.